internal/providers/github/app/app.go

```diff
@@ -84,20 +85,26 @@ func NewGitHubAppProvider(
 
 	ghClient := gogithub.NewClient(tc)
 
-	if config.Endpoint != "" {
-		parsedURL, err := url.Parse(config.Endpoint)
+	if providerConfig.Endpoint != "" {
+		parsedURL, err := url.Parse(providerConfig.Endpoint)
 		if err != nil {
 			return nil, err
 		}
 		ghClient.BaseURL = parsedURL
 	}
 
+	appName := appConfig.AppName
+	userId := appConfig.UserID
+
+	// Alternatively user viper to get the values directly
```

Comment:
We also have the option of getting the values directly from viper but that makes testing harder.